src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -52,34 +57,33 @@ export const useSwapper = () => {
     Trade<KnownChainIds>,
   ]
   const adapterManager = useChainAdapters()
-  const [swapperManager] = useState<SwapperManager>(() => {
-    const manager = new SwapperManager()
-    return manager
-  })
+  const [swapperManager] = useState<SwapperManager>(() => new SwapperManager())
 
   useEffect(() => {
     if (!adapterManager || !swapperManager) return
 
     const web3 = getWeb3Instance()
 
-    ;(async () => {
-      // const midgardUrl = getConfig().REACT_APP_MIDGARD_URL
-      // const thorSwapper = new ThorchainSwapper({
-      //   midgardUrl,
-      //   adapterManager,
-      //   web3,
-      // })
-      // await thorSwapper.initialize()
-      // swapperManager.addSwapper(SwapperType.Thorchain, thorSwapper)
-
-      const zrxSwapper = new ZrxSwapper({
-        web3,
-        adapter: adapterManager.get('eip155:1') as unknown as ethereum.ChainAdapter,
-      })
-
-      await zrxSwapper.initialize()
```

Comment:
LGTM https://github.com/shapeshift/lib/blob/092255cbde52abab5e6a14ed82781eb70560dd09/packages/swapper/src/swappers/zrx/ZrxSwapper.ts#L42-L44

What do we think about making initialize optional in the base `Swapper` class? Or even better, to discriminate based on the swapper type, so that it's `never` for `ZrxSwapper` ?

Reply:
Agree! Made optional in https://github.com/shapeshift/lib/pull/859.

It discriminates nicely in my swapper refactor PR: https://github.com/shapeshift/lib/pull/850

<img width="665" alt="Screen Shot 2022-07-07 at 2 35 42 pm" src="https://user-images.githubusercontent.com/97164662/177692103-d728ea22-aa58-4248-a989-a99a45e2b90a.png">